pkg/controller/statefulset/stateful_set.go

```diff
@@ -149,8 +161,9 @@ func (ssc *StatefulSetController) Run(ctx context.Context, workers int) {
 
 	defer ssc.queue.ShutDown()
 
-	klog.Infof("Starting stateful set controller")
-	defer klog.Infof("Shutting down statefulset controller")
+	logger := klog.LoggerWithName(klog.FromContext(ctx), "StatefulSet")
```

Comment:
Is there a way to avoid defining multiple `logger` and passing it everywhere but can be defined at the beginning of `statefulset` controller itself?

Reply:
We could put a logger into `ssc *StatefulSetController`, but I am not a fan of that: it then becomes much too easy to write code which should log something that got passed down the call chain and doesn't because it uses this `ssc.logger` instance.

Reply:
And I have seen exactly that error being made elsewhere...